src/service/xhr-impl.js

```diff
@@ -106,6 +113,15 @@ export class Xhr {
    * @private
    */
   fetch_(input, init) {
+    if (this.ampdoc_) {
+      // TODO(@jridgewell, #9048): We can alternatively _always_ wait on
+      // #whenFirstVisible.
+      if (Math.random() < .01 && !viewerForDoc(this.ampdoc_).hasBeenVisible()) {
```

Comment:
To follow up on this from above. I definitely agree with @choumx - this signal will get super noisy and even with 1% error rate it would be overwhelming. Sorry for missing it at first. We can already define which requests are "good" or "bad" . The only requests we need to report here are URLs with `url.origin != window.location.origin`. Same-origin requests are explicitly allowed and have 0 risk of any kind associated with them since we just pulled the document itself from that origin.